common/quotas/multistageratelimiter.go

```diff
@@ -20,32 +20,53 @@
 
 package quotas
 
+import "golang.org/x/time/rate"
+
 // MultiStageRateLimiter indicates a domain specific rate limit policy
 type MultiStageRateLimiter struct {
-	domainLimiters *Collection
-	globalLimiter  Limiter
+	// "user"-request limits, e.g. for start, signal, query.
+	domainUserLimiters *Collection
+	// "worker"-request limits, currently just polling.  task-responses are not limited.
+	domainWorkerLimiters *Collection
+
+	globalUserLimiter   Limiter
+	globalWorkerLimiter Limiter
 }
 
 // NewMultiStageRateLimiter returns a new domain quota rate limiter. This is about
 // an order of magnitude slower than
-func NewMultiStageRateLimiter(global Limiter, domainLimiters *Collection) *MultiStageRateLimiter {
+func NewMultiStageRateLimiter(globalUser, globalWorker Limiter, domainUser, domainWorker *Collection) *MultiStageRateLimiter {
```

Comment:
Could we maybe add user/worker split as a separate Limiter implementation?

Right now (at least from the arguments) it seems like we have 4 stages. I would suggest adding
`NewSplitLimiter` or something with like:
```
type SplitLimiter struct {
	userLimiter Limiter
	workerLimiter  Limiter
}
func (sl *SplitLimiter) Allow(info Info) bool {
	if info.IsUser {
		return sl.userLimiter(info)
	} else {
		return sl.workerLimiter(info)
	}
}
```

This would add orthogonal splitting concern as a separate implementation. Encapsulating splitted limiters just as multistage encapsules its stages.

Reply:
Yeah, that'd be a pretty simple change, and pulling it out simplifies things a fair bit. Will do 👍